client/ayon_resolve/plugins/create/create_shot_clip.py

```diff
@@ -240,95 +374,105 @@ def create(self, subset_name, instance_data, pre_create_data):
+                )
+                instance.transient_data["track_item"] = track_item            
+                self._add_instance_to_context(instance)
+                clip_instances[creator_id] = instance.data_to_store()
+
+            # insert clip unique ID and created instances
+            # data as track_item metadata, to retrieve those
+            # during collections and publishing phases
+            lib.imprint(
+                track_item,
+                data={
+                    _CONTENT_ID: clip_instances,
+                    "clip_index": item_unique_id,
+                },
             )
-            instance.transient_data["track_item"] = track_item
-            self._add_instance_to_context(instance)
+            track_item.SetClipColor(constants.publish_clip_color)
+            instances.extend(list(clip_instances.values()))
 
-            # self.imprint_instance_node(instance_node,
-            #                            data=instance.data_to_store())
-            instances.append(instance)
         return instances
 
     def collect_instances(self):
         """Collect all created instances from current timeline."""
-        selected_timeline_items = lib.get_current_timeline_items(
-            filter=True, selecting_color=constants.publish_clip_color)
-
+        all_timeline_items = lib.get_current_timeline_items()
         instances = []
-        for timeline_item_data in selected_timeline_items:
+        for timeline_item_data in all_timeline_items:
             timeline_item = timeline_item_data["clip"]["item"]
 
             # get openpype tag data
             tag_data = lib.get_timeline_item_ayon_tag(timeline_item)
             if not tag_data:
                 continue
 
-            instance = CreatedInstance.from_existing(tag_data, self)
-            instance.transient_data["track_item"] = timeline_item
-            self._add_instance_to_context(instance)
+            for creator_id, data in tag_data.get(_CONTENT_ID, {}).items():
+                creator = self.create_context.creators[creator_id]
+                instance = CreatedInstance.from_existing(data, creator)
+                instance.transient_data["track_item"] = timeline_item
+                self._add_instance_to_context(instance)
+                instances.append(instance)
 
         return instances
 
     def update_instances(self, update_list):
-        """Store changes of existing instances so they can be recollected.
-
-        Args:
-            update_list(List[UpdateData]): Gets list of tuples. Each item
-                contain changed instance and it's changes.
-        """
-        for created_inst, _changes in update_list:
-            track_item = created_inst.transient_data["track_item"]
-            data = created_inst.data_to_store()
-            self.log.info(f"Storing data: {data}")
-
-            lib.imprint(track_item, data)
+        """Never called, update is handled via _ResolveInstanceCreator."""
```

Comment:
Handled by the method definition on the parent class? In what way? Would that mean you just wouldn't re-implement it here instead? or?

Reply:
No, sorry my explanation is probably misleading. 
Basically this `CreateShotClip` creator always delegates instance creation to "sub-creators" which are instances of `_ResolveInstanceCreator` but never hold any instances itself.
Which means that while we need `update_instances`, `remove_instances` and such defined to keep it valid creator, those will never be called from the outside. 

Operations will be handled by the "sub" creators which actually generated the instance:
* update/remove of `shot` will be made through  `ResolveShotInstanceCreator(_ResolveInstanceCreator)`
* update/remove of `plate` will be made through  `EditorialPlateInstanceCreator(_ResolveInstanceCreator)`

Hopefully this clarify a bit, let me know how I could rephase those docstring maybe.